File: helloworld_web_jar/src/main/java/org/example/demo/controller/user/UserController.java
package org.example.demo.controller.user;

import org.example.demo.controller.BaseController;
import org.example.demo.dso.service.UserService;
import org.example.demo.model.UserModel;
import org.noear.solon.annotation.Controller;
import org.noear.solon.annotation.Inject;
import org.noear.solon.annotation.Mapping;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * @author noear
 * @since demo 1.0
 */
@Mapping("user")
@Controller
public class UserController extends BaseController {
    static final Logger log = LoggerFactory.getLogger(UserController.class);

    @Inject
    UserService userService;

    @Mapping("test")
    public String test() {
        log.info("test page");
        UserModel user = userService.getUser(1);
        return "Hello " + user.getUserName();
    }
}

File: helloworld_web_jar_kt/src/main/kotlin/com/example/demo/controller/user/UserController.kt
package com.example.demo.controller.user

import com.example.demo.controller.BaseController
import com.example.demo.dso.service.UserService
import com.example.demo.model.UserModel
import org.noear.solon.annotation.Controller
import org.noear.solon.annotation.Inject
import org.noear.solon.annotation.Mapping
import org.slf4j.Logger
import org.slf4j.LoggerFactory

/**
 * @author noear
 * @since demo 1.0
 */
@Mapping("user")
@Controller
class UserController : BaseController() {
    val log: Logger = LoggerFactory.getLogger(UserController::class.java)

    @Inject
    var userService: UserService? = null

    @Mapping("test")
    fun test(): String {
        log.info("test page")
        val user: UserModel? = userService?.getUser(1)
        return "Hello " + user?.userName;
    }
}

